src/coreclr/src/gc/gc.cpp

```diff
@@ -16006,12 +15953,12 @@ int gc_heap::generation_to_condemn (int n_initial,
             generation_free_list_space (youngest_generation) +
             generation_free_obj_space (youngest_generation);
 
-        dd_fragmentation (dynamic_data_of (max_generation + 1)) =
-            generation_free_list_space (large_object_generation) +
+        dd_fragmentation (dynamic_data_of (loh_generation)) = 
```

Comment:
this would need to loop through uoh generations as we need to do this for all the gens that user code can allocate in (because we update their generation_free_list/obj_space). 

also now there's loh_generation which is a number and large_object_generation which is a generation...